main.go

```diff
@@ -139,7 +139,37 @@ type reconciler struct {
 	gvk          config.GroupVersionKind
 	store
 	namespaces []string
-	orgID      string
+	routes     resourceRoutes
+}
+
+type resourceRoutes struct {
+	clusterResources []string
+	namespaceRoutes  map[string][]string
+}
+
+func newResourceRoutes(routes []config.Route) resourceRoutes {
+	cfg := resourceRoutes{
+		clusterResources: []string{},
+		namespaceRoutes:  map[string][]string{},
+	}
+	for _, route := range routes {
+		if route.ClusterResources {
+			cfg.clusterResources = append(cfg.clusterResources, route.OrganizationID)
+		}
+		for _, ns := range route.Namespaces {
+			cfg.namespaceRoutes[ns] = append(cfg.namespaceRoutes[ns], route.OrganizationID)
+		}
+	}
+	return cfg
+}
+
+// targetOrganizations returns target organizations for given request based on config.Routes
+// Resources can be configured to be routed for zero or more organizations
+func (r resourceRoutes) targetOrganizations(req ctrl.Request) []string {
+	if req.Namespace == "" {
+		return r.clusterResources
+	}
+	return append(r.namespaceRoutes[req.Namespace], r.namespaceRoutes["*"]...)
```

Comment:
That took me a second to understand 😅  (but I don't have any concrete ideas to improve this, at least none that actually improve it 😛 )

One thing I realised is that certain orgs could be in there twice, if they specify `Namespaces: ["*", "default"]` for example.

Maybe we want to validate that :)